pkg/scheduler/scheduler.go

```diff
@@ -241,12 +243,19 @@ func (s *Scheduler) assign(ctx context.Context, e *entry) error {
 // If it finds a flavor, also returns any borrowing required.
 func findFlavorForResource(name corev1.ResourceName, val int64, wUsed map[string]int64, cap *capacity.Capacity, spec *corev1.PodSpec) (string, int64) {
 	for _, flavor := range cap.RequestableResources[name] {
+		// We will only check against the flavors labels.
+		selector := flavorSelector(spec, cap.LabelKeys[name])
```

Comment:
optimization opportunity: build the selector before calling `findFlavorForResource`, once for every resource.

Reply:
well, I should just move it outside the loop :)